spylib/request.py

```diff
@@ -164,29 +159,64 @@ def make_service_request(
             ServiceUnavailable: The service returned a 5XX status code
+            Timeout: The service request timed out
             APIException: An unsupported status code was returned
         
         Returns:
             requests.models.Response: A `Response` object including the service's HTTP response
         """
-        headers = kwargs.get("headers", {})
+        if method not in self.METHOD_MAP:
+            raise MethodException
+        
+        # Build our request
+        url = ServiceRequestFactory.urljoin(base_url, path)
+        
+        # Keep a copy of the kwargs around for retries
+        original_kwargs = copy.deepcopy(kwargs)
+        
+        # Pop headers from kwargs so we don't pass it to requests twice
+        headers = kwargs.pop('headers', {})
         if self.access_token:
             headers.update({"Authorization": "Bearer %s" % self.access_token})
-
-        url = ServiceRequestFactory.urljoin(base_url, path)
-
+        
+        # Build keyword arguments that get passed to the specific requests library function
         if method == "GET":
-            resp = get(url, params=payload, headers=headers, timeout=timeout, **kwargs)
-        elif method == "DELETE":
-            resp = delete(url, headers=headers, timeout=timeout, **kwargs)
-        elif method == "POST":
-            resp = post(url, json=payload, headers=headers, timeout=timeout, **kwargs)
-        elif method == "PATCH":
-            resp = patch(url, json=payload, headers=headers, timeout=timeout, **kwargs)
-        elif method == "PUT":
-            resp = put(url, json=payload, headers=headers, timeout=timeout, **kwargs)
+            # Pop params from kwargs so we don't pass it to requests twice
+            params = payload or {}
+            params.update(kwargs.pop('params', {}))
+            additional_kwargs = {"params": params}
+        elif method in ["POST", "PATCH", "PUT"]:
+            # Pop json from kwargs so we don't pass it to requests twice
+            json = payload or {}
+            json.update(kwargs.pop('json', {}))
+            additional_kwargs = {"json": json}
         else:
-            raise MethodException
+            additional_kwargs = {}
+        
+        func = self.METHOD_MAP[method]
+        try:
+            resp = func(
+                url,
+                headers=headers,
+                timeout=timeout,
+                **kwargs,
+                **additional_kwargs
+            )
+        except RequestsTimeout:
+            # Retry if we can, otherwise throw an internal exception
+            if retry_count > 0:
+                return self.make_service_request(
+                    base_url,
+                    path,
+                    method=method,
+                    payload=payload,
+                    timeout=timeout,
+                    retry_count=retry_count - 1,
+                    retry_on_401_403=retry_on_401_403,
+                    **original_kwargs
```

Comment:
Cool, I think this makes sense. We treat a timeout as a normal retry, and retry if we have a retry available. Otherwise, we terminate. 
Sounds good to me.